microcloud/test/includes/microcloud.sh

```diff
@@ -567,11 +567,14 @@ cluster_reset() {
             microceph.ceph osd pool rm \${pool} \${pool} --yes-i-really-really-mean-it
           done
 
-          for pool in \$(microceph.ceph osd ls) ; do
-            microceph.ceph osd out \${pool}
-            microceph.ceph osd down \${pool} --definitely-dead
-            microceph.ceph osd purge \${pool} --yes-i-really-mean-it --force
-            microceph.ceph osd destroy \${pool} --yes-i-really-mean-it --force
+          # OSDs may re-up themselves before we call destroy, so retry until they are all actually gone.
```

Comment:
@masnax please can you ask the microceph team about this?

Reply:
Sure I'll discuss with them. I don't think this should be a blocker because this whole section is a bit hacky and needs an environment flag to enable. In this section, rather than restoring from snapshot and waiting for the VM agent, we manually disable the snaps, wipe their state directories, clear out any stray modules and processes, and then re-enable the snaps.

It's convenient as it speeds up local testing, but far less clean than just restoring from snapshot, which is the default behaviour of the test suite.

Reply:
This loop's been removed now in favour of following the exact process as MicroCeph when removing disks (but without the slow validation steps).